### PR TITLE
Pen 1313 image path fix

### PR DESCRIPTION
**Pull request by brentmiller1973**
## Description
This solution attempts to solve two issue:
1) Remove the leading forward slash from the image uri to prevent googlebot from attempting to crawl the string as a valid url.
2) Compress the format and quality params to save space in the bundle

## Jira Ticket
- https://arcpublishing.atlassian.net/browse/PEN-1313

## Acceptance Criteria
See Effect Of Changes

## Test Steps
- Run this test:  blocks/resizer-image-block/getResizedImageData.test.js

### Before
When viewing the source of the page, search for the text "resized_params" before the fix the object should look like:

"resized_params": {
    "84x56": "/Zl9GCeY1h_VWYVOhDEUuZLth_I8=filters:format(jpg):quality(70)/",
    "105x70": "/PP7bEkuCBQ9Dv03Ok-8caaFoSI0=filters:format(jpg):quality(70)/",
}

### After
With these changes, the "resized_params" in the source should look like:

"resized_params": {
    "84x56": "Zl9GCeY1h_VWYVOhDEUuZLth_I8=filters:f=jpg:q=70/",
    "105x70": "PP7bEkuCBQ9Dv03Ok-8caaFoSI0=filters:f=jpg:q=70/",
}

## Dependencies or Side Effects
See https://github.com/WPMedia/engine-theme-sdk/pull/97

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps above are working
- [x ] Confirmed there are no linter errors
- [ x] Confirmed this PR has reasonable code coverage
  - [ x] Confirmed this PR has unit test files
  - [x ] Ran `npm test`, made sure all tests are passing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "lint:fix": "npm run lint -- --fix",
     "storybook": "start-storybook",
     "test": "jest",
+    "test:watch": "npm run test -- --watchAll",
```
Comment (JackHowa) [inline package.json:20]: 👍

File: blocks/resizer-image-block/index.js [hunk 2]
```diff
@@ -138,6 +137,15 @@ export const getResizerParams = (
   // todo: use webp for next gen images spike
   // getParamsByFormat('webp', output);
   getParamsByFormat('jpg', output);
+  // Going to clean the data by stripping off leading slash
+  // and compressing format and quality notation
+  // Since this code is only supporting jpg and filterQuality of 70,
+  // we will just assume as much for now.
+  Object.keys(output).forEach((key) => {
+    if (output[key]) {
```
Comment (JackHowa) [inline blocks/resizer-image-block/index.js:148]: what is this for? the existence check?

File: blocks/resizer-image-block/getResizedImageData.test.js [hunk 3]
```diff
@@ -197,13 +197,13 @@ describe('get resized image data helper on the server-side', () => {
 
     // if the same resizer key is used this won't change
     expect(filterValues).toEqual([
-      '/JH9YE5rXU3WoFv5vs342UvuC1gA=filters:format(jpg):quality(70)/',
-      '/oVCzXp5WcaNYOq9omFEyWoMCo78=filters:format(jpg):quality(70)/',
-      '/31mmYPove0AN7aFn_vwKJ0qhZeI=filters:format(jpg):quality(70)/',
-      '/TkU_4vX7gHlX4WBLLwCDMrLuQj4=filters:format(jpg):quality(70)/',
+      'JH9YE5rXU3WoFv5vs342UvuC1gA=filters:f=jpg:q=70/',
```
Comment (JackHowa) [inline blocks/resizer-image-block/getResizedImageData.test.js]: wait, did we want to default to the jpg format? so as not to trigger the googlebot jpg check?

**Comment by JackHowa** [comment]
@visionbegin same here, draft or ready for review my man?

File: blocks/resizer-image-block/index.js (hunk 2 above)
Comment (brentmiller1973) [reply blocks/resizer-image-block/index.js:148]: Defensive coding on the if statement, there could be a case where the value of the key is null.

File: blocks/resizer-image-block/getResizedImageData.test.js (hunk 3 above)
Comment (brentmiller1973) [reply blocks/resizer-image-block/getResizedImageData.test.js]: There is no other format supported at this point in blocks/resizer-image-block/index.js

**Comment by brentmiller1973** [comment]
@JackHowa Ready to go now.for review...I wanted to add the compression param to properly let src/components/Image/thumbor-image-url.ts if its dealing with a compressed/altered string

File: blocks/resizer-image-block/index.js (hunk 2 above)
Comment (JackHowa) [reply blocks/resizer-image-block/index.js:148]: is there a case where that's true? I don't think that should ever happen

File: blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ jest.mock('fusion:properties', () => (jest.fn(() => ({
 describe('The search results', () => {
   describe('renders one list item correctly', () => {
     const element = oneListItem.data[0];
-    if (element?.promo_items?.basic) {
+    if (element?.['promo_items']?.basic) {
```
Comment (JackHowa) [inline blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx]: is there a reason for this change other than style preference? It doesn't seem related?

File: package.json (hunk 1 above)
Comment (JackHowa) [reply package.json:20]: good work here

File: blocks/resizer-image-block/index.js (hunk 2 above)
Comment (JackHowa) [reply blocks/resizer-image-block/index.js:148]: In my mind you can remove this

File: blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx (hunk 4 above)
Comment (JackHowa) [reply blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx]: You can remove this style preference as well unless it was deliberate

**Review by JackHowa — CHANGES_REQUESTED** [review]

File: blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx (hunk 4 above)
Comment (brentmiller1973) [reply blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx]: It doesn't show here, but `element?.promo_items?.basic` required an eslint exception

File: blocks/resizer-image-block/index.js (hunk 2 above)
Comment (brentmiller1973) [reply blocks/resizer-image-block/index.js:148]: There is a test case where it fails without this check

**Comment by brentmiller1973** [comment]
@JackHowa I made some responses to your concerns.

File: blocks/resizer-image-block/index.js (hunk 2 above)
Comment (brentmiller1973) [reply blocks/resizer-image-block/index.js:148]: Without that check, it fails in the test "the author api content source block › .transform › should return author with resized params populated"

Comment (JackHowa) [reply blocks/resizer-image-block/index.js:148]: oh right, I see the test case now: ```resized_params: { '158x89': undefined },```

not wildly sure how that could happen, but looks like Matt did it deliberately

**Comment by JackHowa** [comment]
seeing this linting error: 

```
/Users/howardj/sites/fusion-news-theme-blocks/blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx
   17:19  error  ["promo_items"] is better written in dot notation  dot-notation
   85:19  error  ["promo_items"] is better written in dot notation  dot-notation
  125:19  error  ["promo_items"] is better written in dot notation  dot-notation
```

are you running `npm run lint`? I'd recommend `rm -rf node_modules/ && npm i && npm run lint` if you have older node modules

File: blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx (hunk 4 above)
Comment (JackHowa) [reply blocks/search-results-list-block/features/search-results-list/_children/search-result.test.jsx]: weird that it would cause an eslint error. I'm seeing     `if (element?.['promo_items']?.basic) {` cause an exception when I run `rm -rf node_modules/ && npm i && npm run lint`

**Comment by JackHowa** [comment]
weird it looks like this happened when reviewing another pr https://github.com/WPMedia/fusion-news-theme-blocks/pull/461/commits/eff3642c5b37a3584660be68b71a1d0c0859b971

**Review by JackHowa — APPROVED** [review]

**Comment by JackHowa** [comment]
created issue for placeholder for runbook: https://github.com/WPMedia/fusion-news-theme-blocks/issues/469
